source/adios2/helper/adiosXML.cpp

```diff
@@ -122,49 +155,43 @@ void InitXML(const std::string configXML, const MPI_Comm mpiComm,
     int mpiRank;
     MPI_Comm_rank(mpiComm, &mpiRank);
     std::string fileContents;
-    unsigned long long len;
 
     // Read the file on rank 0 and broadcast it to everybody else
     if (mpiRank == 0)
     {
         fileContents = FileToString(configXML);
-        len = static_cast<unsigned long long>(fileContents.size());
-    }
-    MPI_Bcast(&len, 1, MPI_UNSIGNED_LONG, 0, mpiComm);
-    if (mpiRank != 0)
-    {
-        fileContents.resize(len);
     }
-    MPI_Bcast(const_cast<char *>(fileContents.data()), len, MPI_CHAR, 0,
-              mpiComm);
+
+    fileContents = BroadcastString(fileContents, mpiComm);
 
     pugi::xml_document doc;
     auto parse_result = doc.load_buffer_inplace(
         const_cast<char *>(fileContents.data()), fileContents.size());
-    if (!parse_result)
+
+    if (debugMode)
     {
-        if (debugMode)
+        if (!parse_result)
         {
             throw std::invalid_argument(
-                std::string("ERROR: XML: Parse error: ") +
-                parse_result.description());
+                "ERROR: XML: parse error in file " + configXML +
+                " description: " + std::string(parse_result.description()) +
+                ", in call to ADIOS constructor\n");
         }
-        return;
     }
 
-    pugi::xml_node configNode = doc.child("adios-config");
-    if (!configNode)
+    const pugi::xml_node configNode = doc.child("adios-config");
+
+    if (debugMode)
     {
-        if (debugMode)
+        if (!configNode)
         {
             throw std::invalid_argument(
-                "ERROR: XML: No <adios-config> element found");
+                "ERROR: XML: No <adios-config> element found in file " +
+                configXML + ", in call to ADIOS constructor\n");
         }
-        return;
     }
 
-    ios.clear();
-    for (pugi::xml_node ioNode : configNode.children("io"))
+    for (const pugi::xml_node ioNode : configNode.children("io"))
```

Comment:
It has no effect, this is only called from ADIOS constructors, IOs are always empty at that stage.